src/state/slices/stakingDataSlice/selectors.ts

```diff
@@ -113,6 +149,24 @@ export const selectUnbondingEntriesByAccountSpecifier = createDeepEqualOutputSel
   }
 )
 
+export const selectAllUnbondingsEntriesByDenom = createDeepEqualOutputSelector(
+  selectStakingDataByAccountSpecifier,
+  selectDenom,
+  (stakingData, denom): Record<string, chainAdapters.cosmos.UndelegationEntry[]> => {
+    if (!stakingData || !stakingData.undelegations) return {}
+
+    return stakingData.undelegations.reduce((acc, { validator, entries }) => {
+      if (!acc[validator.address]) {
+        acc[validator.address] = []
+      }
+
+      acc[validator.address].push(...entries.filter(x => ASSET_ID_TO_DENOM[x.assetId] === denom))
```

Comment:
[gomesalexandre](https://github.com/gomesalexandre) [yesterday](https://github.com/shapeshift/web/pull/1356/files#r840057910)
Why the spread?

Array.prototype.filter() is non-mutative and already returns a (shallow) copy in memory (as in, you get a pointer to a whole new array, but each of the key points to the existing references)
Array methods like map and filter also give you a whole new array in memory, but in the same shallow fashion in the case of filter
I don't see the need to duplicate the memory copy of the array both in filter and in the spread.

Reply:
[chxash](https://github.com/chxash) [15 hours ago](https://github.com/shapeshift/web/pull/1356/files#r840360644)
I used a spread because I need all elements to be added separately to the accumulator array, not just having a reference being added by the push. I could use a concat also but I think it would create a new array each time.

Reply:
I dont mind the spread here. Its a nice way for peace of mind to be sure something i a copy without being 100% sure about how the other functions work

Reply:
> I could use a concat also but I think it would create a new array each time.

Spread creates a new array every time and pushing onto an array will require a re-allocation of a new array as the size gets bigger.

Are you sure that `flatMap` isn't more what you're looking for?